### PR TITLE
Huge Scrubbers can now be connected to ports.

### PR DESCRIPTION
**Pull request by Buggy123**
See title. Wrenching them will now, in addition to anchoring/unanchoring them, connect or disconnect them from a port underneath them if it exists.

🆑 Buggy:
tweak: Nanotrasen has updated the design of their extra-large portable air scrubbing equipment. They can now be connected to standard atmospheric ports.
/:cl:

Can't test it for a few hours so no speedmerge please.

### CODE REVIEW

**Comment by Ghilker** [comment]
instead of making them connect to a "connector", could you make them act like a machine and connect directly to pipes? like a freezer/heater but movable

**Comment by Dennok** [comment]
Omg. This can be done in 3 line change.

**Comment by Buggy123** [comment]
That's doable but its fairly complex, because I'd need to change their parent class, and then I'd basically have to copypaste all of the portable scrubber code over to them or make it a heavily modified non-portable scrubber which is also portable...

While I can do it, i'm not sure if it's practical when I currently can't really do much of any spritework and am mostly limited to the webeditor.

**Comment by zxaber** [comment]
It's more convenient that they connect to ports anyway, seeing as atmos has two ports in the hall for people to use to dump waste through.

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 1]
```diff
@@ -165,8 +165,36 @@
...
-	if(default_unfasten_wrench(user, W))
-		if(!movable)
-			on = FALSE
+	if((W.tool_behaviour == TOOL_WRENCH) && !(machine_stat & BROKEN))
+		if(anchored)
+			user.visible_message( \
+				"[user] unanchors [src].", \
+				"<span class='notice'>You unanchor [src] [connected_port ? "and unfasten it from the port underneath it." : "."]</span>", \
+				"<span class='hear'>You hear a ratchet.</span>")
+			investigate_log("was unanchored [connected_port ? "and disconnected from [connected_port] " : NULL]by [key_name(user)].", INVESTIGATE_ATMOS)
```
Comment (spookydonut) [inline code/modules/atmospherics/machinery/portable/scrubber.dm]:
```suggestion
			investigate_log("was unanchored [connected_port ? "and disconnected from [connected_port] " : null]by [key_name(user)].", INVESTIGATE_ATMOS)
```

**Comment by Buggy123** [comment]
begone, stalethot

**Comment by Buggy123** [comment]
Finally got around to testing it, it works.

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 2]
```diff
@@ -165,8 +165,36 @@
 			scrub(AT.return_air())
 
 /obj/machinery/portable_atmospherics/scrubber/huge/attackby(obj/item/W, mob/user)
-	if(default_unfasten_wrench(user, W))
-		if(!movable)
-			on = FALSE
+	if((W.tool_behaviour == TOOL_WRENCH) && !(machine_stat & BROKEN))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/atmospherics/machinery/portable/scrubber.dm]: flip the logic so you're early returning here and not needing the `else return` at the bottome.

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
doesn't the parent already do something similar?

**Comment by Dennok** [comment]
why

**Comment by optimumtact** [comment]
stale

**Pull request closed without merging**